src/types/union.md

```diff
@@ -24,5 +24,17 @@ The memory layout of a `union` is undefined by default (in particular, fields do
 *not* have to be at offset 0), but the `#[repr(...)]` attribute can be used to
 fix a layout.
 
+r[type.union.value]
+A value of a union type consists of a sequence of bytes, corresponding to each [value byte][type.struct.value.value-bytes]. The value bytes of a union are represented exactly. Each [padding byte][type.struct.value.padding] is set to uninit when encoded.
```

Comment:
Given that the fields of a union do not have to be structs, it is odd to refer to `type.struct` here. This definition requires there to be a general notion of "padding bytes in any type", and each type needs to say what its padding bytes are.